### PR TITLE
dm-4594 add autohide to pb events

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-4594

## Description - what does this code do?
Adds optional setting for Event components in PageBuilder that when enabled will hide the event once it's end date  has passed (or start date in the absence of an end date).

If all events adjacent to each other are set to auto-hide it will render a message on the page indicating that all events have passed. Note: this message will not display for the group if any event in a group has passed but is not marked to auto-hide.

## Testing done - how did you test it/steps on how can another person can test it 

1. Run the migration
2. Create a pagebuilder page
3. Create a header component followed by 3 events below it, set the events with start and end dates in the future and check the "AUTO-HIDE ONCE PASSED" box (screenshot 1)
4. Create another header component followed by 3 more events below it, set the events with start and end dates in the past with the "AUTO-HIDE ONCE PASSED" unchecked.
5. View the page and verify the two groups of events appear under their respective headers with the pagination working as expected (displays the first 2 with the "Load morre" button below that reveals the third event for the group. (screenshot 2)
6. Go back into the editor and change the end date for the first event to be in the past.
7. Verify the page renders with only the 2 events that have not passed in the first group, the event that you updated should not be displayed and there should be no "Load more" button under the group. The second group of events should be unaffected. (screenshot 3)
8. Go back and update the same event but remove the end date and set the start date in the past, repeat step 6.
9. Go back and update the same event but remove the start date.
10. Repeat step 4, all events for both groups should again appear with functional pagination.
11. Go back and update all the events in the first group to have start dates in the past
12. View the page again and verify that below the first header there is text that reads "Please check back for more events soon!" in place of the events, and that the other group of events are unaffected. (screenshot 4)

## Screenshots, Gifs, Videos from application (if applicable)

New checkbox for auto-hiding an event in pb edit:
Screenshot 2024-03-27 at 4 28 10 PM

Upcoming events marked as auto-hide but have not yet passed:
Screenshot 2024-03-27 at 2 55 29 PM

Upcoming events marked as auto-hide and one has passed:
Screenshot 2024-03-27 at 4 11 12 PM

Upcoming events marked as auto-hide and all have passed:
Screenshot 2024-03-27 at 2 56 56 PM

## Acceptance criteria
- [ ]

## Definition of done
- [x] Unit tests written (if applicable)
- [x] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating JIRA issue
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: spec/features/pages/event_component_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    expect(page).to have_content(@h3.subtopic_title)
+    expect(page).to have_content("Please check back for more events soon!")
+    expect(page).not_to have_content(@e7.title)
+    expect(page).not_to have_content(@e8.title)
+    expect(page).not_to have_content(@e9.title)
+    expect(page).not_to have_css('.dm-load-more-events-2-btn-container')
+  end
+end
+
```
Comment (PhilipDeFraties) [inline spec/features/pages/event_component_spec.rb:66]: This feature test is very verbose with the fixture creation, but the nuance with adding page_components with correct positions in a way that will mimic practical use was a bit difficult to do in a DRY way so I opted for straight-forward.

File: app/admin/pages.rb [hunk 2]
```diff
@@ -141,7 +142,7 @@
                       img src: "#{component.image_s3_presigned_url}", class: 'maxw-10'
                     end
                   else
-                    li "#{value}: #{component.send(key)}" if component.send(key).present?
+                    li "#{value}: #{component.send(key)}" if (component.send(key) == false || component.send(key).present?)
```
Comment (PhilipDeFraties) [inline app/admin/pages.rb:145]: This change I included so that boolean attrs set to `false` will show up in the top view for the pb components

Comment (camillevilla) [reply app/admin/pages.rb:145]: Nice! Definitely a good add on.

**Review by camillevilla — APPROVED** [review]
Looking good! I just have two small language suggestions for the form input.

**Pull request merged**